File: frontend/src/lib/api/supplier.ts
import { supabase } from '@/lib/supabase'

export interface SupplierFormData {
  company_name: string
  description: string
  categories: string[]
  subcategories: string[]
  specialties: string
  certifications: string[]
  location: string
}

export interface SupplierMatch {
  supplier: {
    id: number
    company_name: string
    description: string
    categories: string[]
    subcategories: string[]
    specialties: string
    certifications: string[]
    location: string
  }
  score: number
  category_match: boolean
  subcategory_match: boolean
}

export async function createSupplierProfile(data: SupplierFormData) {
  const { data: session } = await supabase.auth.getSession()
  if (!session) {
    throw new Error('Not authenticated')
  }

  const response = await fetch('/api/v1/suppliers', {
    method: 'POST',
    headers: {
      'Content-Type': 'application/json',
      'Authorization': `Bearer ${session.access_token}`
    },
    body: JSON.stringify(data)
  })

  if (!response.ok) {
    const error = await response.json()
    throw new Error(error.detail || 'Failed to create supplier profile')
  }

  return response.json()
}

export async function updateSupplierProfile(supplierId: number, data: Partial<SupplierFormData>) {
  const { data: session } = await supabase.auth.getSession()
  if (!session) {
    throw new Error('Not authenticated')
  }

  const response = await fetch(`/api/v1/suppliers/${supplierId}`, {
    method: 'PUT',
    headers: {
      'Content-Type': 'application/json',
      'Authorization': `Bearer ${session.access_token}`
    },
    body: JSON.stringify(data)
  })

  if (!response.ok) {
    const error = await response.json()
    throw new Error(error.detail || 'Failed to update supplier profile')
  }

  return response.json()
}

export async function getMatchingSuppliers(rfqId: number) {
  const { data: session } = await supabase.auth.getSession()
  if (!session) {
    throw new Error('Not authenticated')
  }

  const response = await fetch(`/api/v1/rfq/${rfqId}/matches`, {
    headers: {
      'Authorization': `Bearer ${session.access_token}`
    }
  })

  if (!response.ok) {
    const error = await response.json()
    throw new Error(error.detail || 'Failed to get matching suppliers')
  }

  return response.json() as Promise<SupplierMatch[]>
}

File: backend/app/schemas/supplier.py
from pydantic import BaseModel, Field
from typing import List, Optional

class SupplierBase(BaseModel):
    company_name: str = Field(..., min_length=2, max_length=100)
    description: str = Field(..., min_length=20)
    categories: List[str]
    subcategories: List[str]
    specialties: str = Field(..., min_length=20)
    certifications: Optional[List[str]] = []
    location: str

class SupplierCreate(SupplierBase):
    pass

class SupplierUpdate(SupplierBase):
    company_name: Optional[str] = None
    description: Optional[str] = None
    categories: Optional[List[str]] = None
    subcategories: Optional[List[str]] = None
    specialties: Optional[str] = None
    certifications: Optional[List[str]] = None
    location: Optional[str] = None

class Supplier(SupplierBase):
    id: int
    user_id: int
    embedding: Optional[List[float]] = None

    class Config:
        from_attributes = True

class SupplierMatch(BaseModel):
    supplier: Supplier
    score: float
    category_match: bool
    subcategory_match: bool

    class Config:
        from_attributes = True
